### PR TITLE
Remove „public annotation types“ from the list

### PR DESCRIPTION
**Pull request by lingocoder**
### TL;DR

What are some examples of „*public annotation types*“?

### Assumptions

The „*Recognizing API and implementation dependencies*“ section advises that „*public annotation types*“ should be recognized as API dependencies.

My proposed file change is based on the following assumption of what is meant by „*public annotation types*“:  Annotations ***used*** by a library, that consumers of that library would be required to include in their (*consumers'*) class path.

### Context

I'm developing [*an application that programmatically analyzes the public API of a given library*](http://bit.ly/AbiImpl), It will offer the Gradle community a way to detect whether their public APIs expose any of the items included in this page's list of things suggested as API dependencies. 

Using reflection, the application does detect annotations used in libraries if those annotations are implemented with the [*`RetentionPolicy.RUNTIME`*](http://bit.ly/RntPlcRT).

My current understanding of annotations is: In order for consumers to be compiled against a library that ***uses*** annotations, it is never necessary for consumers of the library to have those same annotations in their (consumers') compile-time class path. Not even for annotations with *`RUNTIME`* retention.

I am suggesting that there are no such annotations. Therefore, that bullet point in question should be removed. If such annotations do exist, then instead of that item being removed, it would improve the understand-ability of the guidelines if examples of „*public annotation types*“ were given somewhere in this page. 

If it is a case of me simply not fully understanding the topic, then an answer given [*to the question posed to the wider Gradle user community*](http://bit.ly/AintNone) would help many others who may have similar gaps in their understanding.

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by big-guy** [comment]
I think your PR got messed up a bit--I don't see any changes.  I replied to your forum post, I think the line makes sense, but could/should be explained more (PRs are welcome)
